apps/src/templates/studioHomepages/MarketingAnnouncementBanner.jsx

```diff
@@ -8,68 +8,61 @@ import shapes from './shapes';
+  //
+  // Note that Optimize supports multiple experiments on the same page, but
+  // there is no supported way to determine which elements on the page were
+  // modified by which experiments. We assume that there will be only one
+  // Personalization experiment on a given page.
+  const [activeExperimentId, setActiveExperimentId] = useState(null);
   useEffect(() => {
-    if (window['optimizely']) {
-      const optimizelyUtils = window['optimizely'].get('utils');
-      // When modifications are made to the banner through optimizely, check whether
-      // this version of the banner has been dismissed by the teacher
-      optimizelyUtils.observeSelector(
-        `#${bannerRef.current.id}`,
-        checkShouldDisplayBanner
-      );
+    // Sites that use Google Analytics typically have a gtag() function which
+    // is used to push data to Google Ad Platform's data layer. Our site does not
+    // currently globally define this function so we define a private copy here.
+    function _gtag() {
+      if (window.dataLayer) {
+        window.dataLayer.push(arguments);
+      }
     }
-    checkShouldDisplayBanner();
+
+    _gtag('event', 'optimize.callback', {
+      callback: (variant, experimentId) => {
+        // Personalization experiments have variant equal to ''.
+        if (variant === '') {
+          setActiveExperimentId(experimentId);
+        }
+      }
+    });
   }, []);
 
-  const checkShouldDisplayBanner = () => {
+  const shouldDisplayBanner = () => {
     const bannerKey = getLocalStorageBannerKey();
     const displayBanner = tryGetLocalStorage(bannerKey, true);
-    if (displayBanner === 'false') {
-      setDisplayBanner(false);
-    }
+    return !(displayBanner === 'false');
   };
 
   const getLocalStorageBannerKey = () => {
     let bannerId = announcement.id;
-
-    const optimizelyId = getOptimizelyModifiedElementId();
-    if (optimizelyId) {
-      bannerId = optimizelyId;
+    if (activeExperimentId) {
+      bannerId = activeExperimentId;
     }
-
     return `display-announcement-${bannerId}`;
   };
 
-  const getOptimizelyModifiedElementId = () => {
-    const allBannerElements = bannerRef.current.querySelectorAll('*');
-
-    const getOptlyDataAttrKey = element => {
-      return Object.keys(element.dataset).find(key => key.includes('optly'));
-    };
-
-    // Finds an element that was modified by optimizely if one exists
-    const optlyModifiedElement = [...allBannerElements].find(el =>
-      getOptlyDataAttrKey(el)
-    );
-
-    if (optlyModifiedElement) {
-      // Returns the optimizely data attribute key for the changed element
-      // will be something like optly-0ef57bf5F12b-4290A4dbA1de95a9b5cd
-      return getOptlyDataAttrKey(optlyModifiedElement);
-    }
-  };
-
   const onDismiss = () => {
     const bannerKey = getLocalStorageBannerKey();
     trySetLocalStorage(bannerKey, false);
-    setDisplayBanner(false);
+    forceUpdate();
```

Comment:
Could you explain what this `forceUpdate` is used for? I'm not sure I understand

Reply:
Without the `forceUpdate` here, there are no state changes that will trigger a re-render.  I can also try putting back the function that calls setDisplayBanner() if you think that's more straightforward.  (I had changed this at some point because it seemed to work better with the 'gtmProgressiveApplied' attribute check which ended up not making it in in this PR.)

Reply:
It's a pattern that's somewhat discouraged but I thought it was ok to use here because this callback updates external state which needs to be evaluated on the next render.

Reply:
I see, you're forcing the component to re-render since the update isn't related to state. This seems like an unexpected pattern, I took a look https://reactjs.org/docs/hooks-faq.html#is-there-something-like-forceupdate it does say 'Try to avoid this pattern if possible'. I'd recommend that we store whether the component is displayed in the state and update that so we don't need to force the component to re-render